client/src/components/BylineMetaBox/BylineProfiles.js

```diff
@@ -131,92 +137,102 @@ class BylineProfiles extends Component {
 
     return (
       <div>
-        <div className="byline-list-controls">
-          <div className="profile-controls">
+        <div className="byline-list-controls components-base-control">
+          <div className="profile-controls components-base-control__field">
             {/* eslint-disable jsx-a11y/label-has-for */}
-            <label htmlFor="profiles_autocomplete">
+            <label
+              className="components-base-control__label"
+              htmlFor="profiles_autocomplete"
+            >
               {window.bylineData.addAuthorLabel}
-              <Autocomplete
-                inputProps={inputProps}
-                items={this.state.searchResults}
-                value={this.state.search}
-                getItemValue={(item) => item.name}
-                wrapperStyle={{ position: 'relative', display: 'block' }}
-                onSelect={(value, item) => {
+            </label>
+            <Autocomplete
+              inputProps={inputProps}
+              items={this.state.searchResults}
+              value={this.state.search}
+              getItemValue={(item) => item.name}
+              wrapperStyle={{ position: 'relative', display: 'block' }}
+              onSelect={(value, item) => {
+                this.setState((state) => ({
+                  search: '',
+                  searchResults: [],
+                  profiles: [
+                    ...state.profiles,
+                    item,
+                  ],
+                }));
+              }}
+              onChange={(event, value) => {
+                clearTimeout(this.delay);
+                this.setState({
+                  search: value,
+                });
+
+                this.delay = setTimeout(() => {
+                  this.doProfileSearch(value);
+                }, 500);
+              }}
+              renderMenu={(children) => (
+                <div className="menu">
+                  {children}
+                </div>
+              )}
+              renderItem={(item, isHighlighted) => (
+                <div
+                  className={`item ${isHighlighted ?
+                    'item-highlighted' : ''}`}
+                  key={item.id}
+                >
+                  {item.name}
+                </div>
+              )}
+            />
+          </div>
+          <div className="freeform-controls components-base-control__field">
+            <label
+              className="components-base-control__label"
+              htmlFor="byline_freeform"
+            >
+              {window.bylineData.addFreeformlabel}
```

Comment:
Shouldn't access the `window` object here - if there is configuration data that gets passed by WP then that should either flow down into this component as a prop or it should be extracted to a function that is responsible for getting a configuration value and interacting with `window`.

Reply:
Looks like we're using localize script here which is feeding the slotfill/components a bunch of data. Will have to think about how best to restructure this. I might open a new branch to wrangle this and a few other improvements and figure out how best to handle this...